pkg/resource/internal/compare/compare.go

```diff
@@ -0,0 +1,117 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package compare implements term operation helpers.
+package compare
+
+import (
+	"strconv"
+	"strings"
+	"unicode"
+)
+
+// GetNumbers returns numbers parsed from left and right params, if any of string to number conversion fails returns false as the 3rd arg.
+func GetNumbers(left, right string) (int64, int64, bool) {
+	numLeft, ok := parseValue(left)
+	if !ok {
+		return 0, 0, false
+	}
+
+	numRight, ok := parseValue(right)
+	if !ok {
+		return 0, 0, false
+	}
+
+	return numLeft, numRight, true
+}
+
+func parseValue(value string) (int64, bool) {
+	value = strings.TrimSpace(value)
+
+	sign := int64(1)
+
+	if string(value[0]) == "-" {
```

Comment:
`strconv.ParseInt` handles negative numbers perfectly fine, I don't think we need to handle it this way